ethcore/src/spec/spec.rs

```diff
@@ -21,55 +21,51 @@ use engine::*;
 use pod_state::*;
 use null_engine::*;
 use account_db::*;
+use super::genesis::Genesis;
+use super::seal::Generic as GenericSeal;
 use ethereum;
-use super::genesis::{Seal as GenesisSeal, Genesis};
-
-/// Convert JSON value to equivalent RLP representation.
-// TODO: handle container types.
-fn json_to_rlp(json: &Json) -> Bytes {
-	match *json {
-		Json::Boolean(o) => encode(&(if o {1u64} else {0})).to_vec(),
-		Json::I64(o) => encode(&(o as u64)).to_vec(),
-		Json::U64(o) => encode(&o).to_vec(),
-		Json::String(ref s) if s.len() >= 2 && &s[0..2] == "0x" && U256::from_str(&s[2..]).is_ok() => {
-			encode(&U256::from_str(&s[2..]).unwrap()).to_vec()
-		},
-		Json::String(ref s) => {
-			encode(s).to_vec()
-		},
-		_ => panic!()
-	}
+use ethjson;
+
+/// Parameters common to all engines.
+#[derive(Debug, PartialEq, Clone)]
+pub struct CommonParams {
+	/// Account start nonce.
+	pub account_start_nonce: U256,
+	/// Frontier compatibility mode limit.
+	pub frontier_compatibility_mode_limit: u64,
```

Comment:
it's also used in `NullEngine` so I assumed that it's a property common to all engines